processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -184,6 +184,12 @@ public void postFilters(List<Filter> postFilters)
   }
 
   @Override
+  public void identity(String identity)
+  {
+    setDimension("identity", identity);
```

Comment:
I'm still against adding any new dimensions, emitted by default, see https://github.com/druid-io/druid/pull/4570#issuecomment-317878173 (and later comments in that thread). Also I committed to contribute something like "full" metrics, with big disclaimer of no backward compatibility: https://github.com/druid-io/druid/pull/4570#issuecomment-321026206.

But if you still want to emit it by default, I don't have much motivation to fight against this.

Reply:
I thought the potential usefulness of the dimension outweighed the concerns you raised in https://github.com/druid-io/druid/pull/4570#issuecomment-317878173, especially since unless you go and specifically set up custom authenticators, the cardinality of the dimension will be 1.

What do you feel about a compromise where this dimension is emitted by default only if the user has authentication configured?

Reply:
@gianm 

>emitted by default only if the user has authentication configured?

What did you have in mind for the "has authentication configured" check?

Reply:
@jon-wei I was thinking having a non-default authenticators list.

Reply:
@gianm my main concert is inability to take out or change the dimension in the future, not inefficiency in this case

Reply:
@gianm that sounds good to me

Reply:
@leventov @jon-wei I changed it to not emit identity by default. Maybe it will change in a future patch but I will keep this one uncontroversial.